cogs/moderation.py

```diff
@@ -492,6 +572,184 @@ async def on_member_kick(self, target, entry):
+                """
+            ),
+        )
+        confirm_embed.add_field(
+            name="Are you sure that you want to send an Emergency Staff Alert for the following reason?",
+            value=reason,
+            inline=False,
+        )
+
+        rules_embed = discord.Embed(
+            color=discord.Color.blurple(),
+            title="Use Cases & Abuse",
+            description=textwrap.dedent(
+                f"""
+                Abuse of this alert system is **strictly prohibited** and **will** result in repercussions if used maliciously. Here are some examples to help understand when and when not to use it, this is not exhaustive:
+                **✅ Acceptable Cases**
+                - Sending NSFW/disturbing content in our server(s)/DMs
+                - Advertising Crosstrading/Distribution of automated scripts in our server(s) that violate our ToS
+                - Malicious/excessive spam in our server(s)
+                - Advertising links to malicious/scam websites in our server(s)/DMs
+                - Extreme Toxicity/Harassment/Trolling
+                - Actively violating any other rule to an excessive extent
+                **<:white_cross_mark:1193650425166045224> Unacceptable Cases**
+                - Suspected autocatching in our server(s)
+                - Server advertisement
+                - Bot outages/bugs/glitches — Please use #bug-reports or ping a Developer in case of emergency
+                - Asking staff to check appeals/applications
+                """
+            ),
+        )
+        rules_embed.set_footer(text="Please use `?report` in unacceptable cases that violate our rules.")
+
+        if not await ctx.confirm(timeout=120, embeds=[confirm_embed, rules_embed], ephemeral=True):
+            ctx.command.reset_cooldown(ctx)
+            return await ctx.send("Aborted.", ephemeral=True)
+
+        alert_embed = discord.Embed(color=discord.Color.red(), title="🚨 Emergency Staff Alert Issued", description="")
+        alert_embed.set_author(
+            name=f"{ctx.author} ({ctx.author.id})",
+            icon_url=ctx.author.display_avatar,
+        )
+        alert_embed.add_field(name="Reason", value=reason, inline=False)
+        view = EmergencyView(ctx)
+        view.add_item(
+            discord.ui.Button(
+                label="Logs",
+                url=f"https://admin.poketwo.net/logs/{ctx.guild.id}/{ctx.channel.id}?before={ctx.message.id+1}",
+            )
+        )
+        view.message = await ctx.reply(role.mention, embed=alert_embed, view=view)
+
+    @emergency.error
+    async def emergency_error(self, ctx: GuiduckContext, error):
+        if isinstance(error, commands.CommandOnCooldown):
+            await ctx.send(
+                f"An Emergency Staff Alert has already been issued recently and is currently on cooldown. Please use `?report` instead if necessary.",
+                ephemeral=True,
+            )
+
+    @emergency.command(name="set-role")
```

Comment:
I would prefer if we could just look for a role called "Emergency Staff Alert" instead of setting one, since that's how Muted works

Reply:
Done, but may need to revisit this as the role is not being mentioned due to the disallowance in the bot constructor.